### PR TITLE
feat(membership): carry effective_at through durable Member timestamps

### PR DESCRIPTION
**Pull request by fahertym**
## What

Implements the #2286 durable membership timestamp contract from #2287 ([`docs/design/membership-durable-timestamp-semantics.md`](docs/design/membership-durable-timestamp-semantics.md)).

Durable membership timestamps for add/remove/freeze/unfreeze now come from a decision-carried `effective_at` rather than node-local wall-clock reads, so two nodes replaying the same governance decision converge on the durable `Member` **logical** state (a normalized durable-state projection) — closing the durability gap #2284 left after making `state_change_hash` a deterministic decision-identity fingerprint.

> **Scope of the convergence claim:** this claims *logical* convergence (typed fields + order-normalized metadata), not raw `save_member` byte-identity. `Member.metadata` is a `HashMap` postcard-encoded as part of the whole struct, so iteration order (hence bytes) is nondeterministic across processes even when contents match. A byte-level guarantee would need a canonical `Member` encoding — a separate concern, out of scope for #2286. The two-node test asserts the §7-sanctioned normalized projection (metadata sorted). Raised by Codex review; addressed in doc §10 + test.

## Design decisions

- `effective_at` is carried on `MembershipEffect::{AddMember, RemoveMember, FreezeMember, UnfreezeMember}` and mirrored on the corresponding kernel-api request structs.
- `UpdateMember` / `UpdateMemberRequest` are **unchanged** — they persist no durable timestamp today (the audit confirmed their wall-clock read feeds only in-memory provenance), so adding one would be new surface, not #2286 repair.
- `freeze_expires_at = effective_at + duration_secs` (saturating).
- The `membership:v2` `state_change_hash` layout is **byte-for-byte unchanged**; a regression test asserts the fingerprint is independent of `effective_at`.
- `Member::new` is left intact for non-governance callers; a new `with_joined_at_secs` seam sets `joined_at` from `effective_at` after construction.
- The in-memory `MembershipProvenance.timestamp` stays node-local audit metadata, separate from durable convergence state.

**Effective-at source: `SystemEvent::ProposalAccepted.decided_at`.** The governance decision's own recorded time — sampled once when the proposal is closed (`actor.rs`, `decided_at: now`), persisted in the close journal (`close_journal.rs`: *"recovery reproduces the same `decided_at` rather than using restart time"*), and gossiped in the event. Every node replaying the decision reads the same value; it is never a per-node clock read, and is unrelated to the deprecated proposal-level delayed-execution field (#282). Threaded via `translate_payload_to_effects` (`apps/governance/src/lib.rs` → `handlers/execution.rs`).

**Compatibility posture: versioned `Option<u64>` + `#[serde(default)]`, not fail-closed.** Live evidence: membership effects are persisted in `ExecutionRecord.effects` and re-decoded on the crash-recovery convergence path (`decision_executor.rs`), so a bare non-`default` field would break recovery of in-flight pre-upgrade decisions. Therefore:
- new decisions always carry `Some(decided_at)` — fully convergent;
- effects serialized before this field decode as `None` (**never** epoch zero — no fabricated protocol time) and the service falls back to the pre-#2286 node-local `current_timestamp_secs()` for the durable write, a path that is explicitly non-convergent and reachable only by legacy effects.

## Files

- `icn-kernel-api/src/effects.rs`, `services.rs` — `effective_at: Option<u64>` on the four effects/requests.
- `icn-core/src/supervisor/governance_executor.rs` — thread effect → request.
- `apps/governance/src/handlers/execution.rs`, `lib.rs` — producer + `decided_at` source.
- `icn-coop/src/types.rs` — `Member::with_joined_at_secs` seam.
- `icn-core/src/services/membership_service.rs` — durable fields use `effective_at`; new tests.
- Test call-site/fixtures updates + `federated_two_node_pilot.rs` durable-record convergence assertion.
- `docs/design/membership-durable-timestamp-semantics.md` — §10 records the two sub-decisions made here.

## Validation

- `cargo test -p icn-kernel-api --lib` → 616 passed.
- `cargo test -p icn-coop --lib` → 164 passed.
- `cargo test -p icn-core --lib services::membership_service` → 14 passed (4 new: joined_at == effective_at; mutation ops persist effective_at + freeze expiry; legacy `None` → local, non-epoch-zero; hash ignores `effective_at`).
- `cargo test -p icn-core --test federated_two_node_pilot` → 6 passed (extended: two-node durable `Member` projection convergence + `joined_at == effective_at`).
- `cargo test -p icn-core --test {effect_group_integration, membership_action_governance_integration, freeze_unfreeze_governance_integration, remove_member_governance_integration}` → 10 / 2 / 3 / 2 passed.
- `cargo fmt --all --check` clean; `cargo clippy -p icn-kernel-api -p icn-coop -p icn-governance-actor -p icn-core --all-targets -- -D warnings` clean; `cargo check --workspace --all-targets` clean; `git diff --check` clean.
- No OpenAPI/SDK impact: the changed kernel-api types carry no `utoipa::ToSchema` derive and appear in neither `docs/api/openapi.generated.yaml` nor the TS SDK.

## Non-claims

No production, pilot, organizer-ready, member-ready, live-federation, or Phase 2 completion claim.
No #1748/#2141 closure.
No #2081/#2080/#2274 progress.
No process-spine change.
No `membership:v2` hash-layout change.
Durable convergence is claimed only for the membership add/remove/freeze/unfreeze path implemented and tested here; legacy (`None`) effects remain intentionally non-convergent.

Refs #2286
Refs #2287
Refs #2284
Refs #2283

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: docs/design/membership-durable-timestamp-semantics.md [hunk 1]
```diff
@@ -324,3 +324,43 @@ implementation PR referencing this contract. The one sub-decision that
...
+## 10. Implementation PR decision (`feat/membership-effective-at-durable-timestamps`)
+
+The implementation PR made the two open sub-decisions this contract left to
+it. Recorded here so the doc stays the audit trail.
+
+**Source of `effective_at` (§5): `ProposalAccepted.decided_at`.** The
+governance decision's own recorded time (`icn-kernel-api/src/events.rs`,
+`SystemEvent::ProposalAccepted.decided_at`) is threaded through
+`translate_payload_to_effects` (`apps/governance/src/lib.rs` →
+`handlers/execution.rs`) into the four durable membership effects. It is
```
[the remark this answers is not in the document]
Comment (fahertym) [reply docs/design/membership-durable-timestamp-semantics.md]: Normalized these path references to the repo-root `icn/apps/...` / `icn/crates/...` style used elsewhere in the doc (commit bf9d2c1f).

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: docs/design/membership-durable-timestamp-semantics.md [hunk 2]
```diff
@@ -324,3 +324,43 @@ implementation PR referencing this contract. The one sub-decision that
...
+sampled exactly once when the proposal is closed
+(`apps/governance/src/actor.rs`, `decided_at: now`), persisted in the close
+journal (`apps/governance/src/close_journal.rs` — "recovery reproduces the
+same `decided_at` rather than using restart time"), and gossiped in the
+event, so every node replaying the decision reads an identical value and
```
[the remark this answers is not in the document]
Comment (fahertym) [reply docs/design/membership-durable-timestamp-semantics.md]: Fixed — this `apps/governance/...` reference now uses the `icn/apps/...` style (commit bf9d2c1f).

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: docs/design/membership-durable-timestamp-semantics.md [hunk 3]
```diff
@@ -324,3 +324,43 @@ implementation PR referencing this contract. The one sub-decision that
...
+never a local clock. It is the membership analogue of the protocol lane's
+decision-carried `effective_at`, and is unrelated to the deprecated
+proposal-level delayed-execution field (#282).
+
+**Compatibility posture (§6): versioned `Option<u64>`, not fail-closed.**
+The audit found that membership effects are persisted in
+`ExecutionRecord.effects` and re-decoded on the crash-recovery convergence
+path (`icn-core/src/supervisor/decision_executor.rs`), so a bare
+non-`default` field would break recovery of in-flight pre-upgrade decisions.
```
[the remark this answers is not in the document]
Comment (fahertym) [reply docs/design/membership-durable-timestamp-semantics.md:352]: Added the missing `icn/crates/` prefix on the `decision_executor.rs` anchor (commit bf9d2c1f).

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: icn/crates/icn-core/tests/federated_two_node_pilot.rs [hunk 4]
```diff
@@ -381,7 +387,40 @@ async fn test_two_node_membership_add_determinism() -> Result<()> {
...
+    // diverged across a seconds boundary. Compare a normalized durable-state
+    // projection that includes the timestamp.
+    let member_a = coop_store_a.get_member("coop-sunrise-pilot", &member_did)?;
+    let member_b = coop_store_b.get_member("coop-sunrise-pilot", &member_did)?;
+
+    let project = |m: &icn_coop::Member| {
+        (
+            m.joined_at,
+            format!("{:?}", m.role),
+            format!("{:?}", m.status),
```
[the remark this answers is not in the document]
Comment (fahertym) [reply icn/crates/icn-core/tests/federated_two_node_pilot.rs:421]: Correct — the raw `Member` bytes aren't identical because `metadata` is a postcard-encoded `HashMap` with nondeterministic iteration order. Narrowed the claim to the normalized durable-state projection (the #2287 §7-sanctioned option): the two-node test now also compares order-normalized (sorted) metadata to prove logical convergence, and doc §10 records that raw byte-identity needs a canonical `Member` encoding (e.g. `BTreeMap`), which is a separate concern out of scope for #2286. Commit bf9d2c1f.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_
